src/flamegpu/simulation/CUDAEnsemble.cu

```diff
@@ -172,6 +185,36 @@ unsigned int CUDAEnsemble::simulate(const RunPlanVector& plans) {
     // Return to device 0 (or check original device first?)
     gpuErrchk(cudaSetDevice(0));
 
+    // If there are no devices left (and mpi is not being used), we need to error as the work cannot be executed.
+#ifdef FLAMEGPU_ENABLE_MPI
+    if (devices.size() == 0 && mpi == nullptr) {
```

Comment:
`mpi==nullptr` will always evaluate `false`?

Reply:
Based on this. 

https://github.com/FLAMEGPU/FLAMEGPU2/blob/cdad83a2dd9c34e2ebefc3a7fbea740180e7a62e/src/flamegpu/simulation/CUDAEnsemble.cu#L66